client/main.lua

```diff
@@ -92,90 +108,151 @@ local function GetDeliveryLocation()
+        id = 'qb_busjob_open_garage_context_menu',
         title = Lang:t('menu.bus_header'),
         options = vehicleMenu
     })
-    lib.showContext('open_busGarage_context_menu')
+    lib.showContext('qb_busjob_open_garage_context_menu')
+end
+
+local function updateZone()
+    if table.type(PlayerData) == 'empty' or PlayerData.job.name ~= 'bus' then
+        if VehicleZone then
+            VehicleZone:remove()
+            VehicleZone = nil
+        end
+        return
+    end
+
+    if not VehicleZone then
+        local inRange = false
+        local shownTextUI = false
+        local inVeh = whitelistedVehicle()
+        VehicleZone = lib.zones.sphere({
+            name = "qb_busjob_bus_main",
+            coords = Config.Location.xyz,
+            radius = 5,
+            debug = false,
+            onEnter = function()
+                inRange = true
+                CreateThread(function()
+                    repeat
+                        Wait(0)
+                        if not inVeh then
+                            if not shownTextUI then
+                                lib.showTextUI(Lang:t('info.busstop_text'))
+                                shownTextUI = true
+                            end
+                            if IsControlJustReleased(0, 38) then
+                                busGarage()
+                                lib.hideTextUI()
+                                shownTextUI = false
+                                break
+                            end
+                        else
+                            if not shownTextUI then
+                                lib.showTextUI(Lang:t('info.bus_stop_work'))
+                                shownTextUI = true
+                            end
+                            if IsControlJustReleased(0, 38) then
+                                if not NpcData.Active or NpcData.Active and not NpcData.NpcTaken then
+                                    if cache.vehicle then
+                                        BusData.Active = false
+                                        DeleteVehicle(cache.vehicle)
+                                        RemoveBlip(NpcData.NpcBlip)
+                                        lib.hideTextUI()
+                                        shownTextUI = false
+                                        resetNpcTask()
+                                        break
+                                    end
+                                else
+                                    QBCore.Functions.Notify(Lang:t('error.drop_off_passengers'), 'error')
+                                end
+                            end
+                        end
+                    until not inRange
+                end)
+            end,
+            onExit = function()
+                lib.hideTextUI()
+                shownTextUI = false
+                inRange = false
+            end
+        })
+    end
```

Comment:
Consider breaking this function up by extracting functions out of it to get rid of the nesting.

Reply:
It's only used once, so I don't see the point in that